### PR TITLE
Update sly striker rule elements to function in V10

### LINKED ISSUE #4466: Feat: Sly Striker doesn't work
**Feat Description:** Your attacks deal more damage, even against creatures that aren't flat-footed.

When you succeed or critically succeed at a Strike against a creature that isn't flat-footed, you also deal 1d6 precision damage. This applies only if you're using a weapon or unarmed attack you could deal sneak attack damage with.

At 14th level, if you would normally deal 3d6 or more sneak attack damage to flat-footed creatures, you deal 2d6 precision damage to creatures that aren't flat-footed.

**Rules:**
{"damageType":"precision","dieSize":"d6","key":"DamageDice","predicate":[{"or":["finesse","agile","ranged"]},{"not":"target:condition:flat-footed"}],"selector":"strike-damage","value":{"brackets":[{"end":10,"value":{"diceNumber":1}},{"start":14,"value":{"diceNumber":2}}]}}

### PR DESCRIPTION
**Pull request by TMunsie**
Corrects predicates
Adds support for rogue dedication
Closes #4466

### CODE REVIEW

File: packs/data/feats.db/sly-striker.json [hunk 1]
```diff
@@ -30,15 +30,23 @@
...
-                            "finesse",
-                            "agile",
-                            "ranged"
+                            "weapon:trait:agile",
+                            "weapon:trait:finesse",
+                            {
+                                "and": [
+                                    "weapon:ranged",
+                                    {
+                                        "not": "weapon:thrown-melee"
```
Comment (TikaelSol) [inline packs/data/feats.db/sly-striker.json:46]:
```suggestion
                                        "not": ["weapon:thrown-melee"]
```

File: packs/data/feats.db/sly-striker.json [hunk 2]
```diff
@@ -62,6 +70,33 @@
...
+                                    }
+                                ]
+                            }
+                        ]
+                    },
+                    {
+                        "not": "target:condition:flat-footed"
```
Comment (TikaelSol) [inline packs/data/feats.db/sly-striker.json:96]:
```suggestion
                                    {
                                        "not": ["weapon:thrown-melee"]
                                    }
                                ]
                            }
                        ]
                    },
                    {
                        "not": ["target:condition:flat-footed"]
```

File: packs/data/feats.db/sly-striker.json (hunk 1 above)
Comment (stwlam) [reply packs/data/feats.db/sly-striker.json:46]: `not` takes a single string (the original is correct)

File: packs/data/feats.db/sly-striker.json (hunk 2 above)
Comment (stwlam) [reply packs/data/feats.db/sly-striker.json:96]: same

**Pull request merged**